flexget/plugins/input/sonarr.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+import requests
+
+from flexget import plugin
+from flexget.event import event
+from flexget.entry import Entry
+
+log = logging.getLogger('sonarr')
+
+
+class Sonarr(object):
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'base_url': {'type': 'string'},
+            'port': {'type': 'number'},
+            'api_key': {'type': 'string'}
+        },
+        'required': ['api_key', 'base_url', 'port'],
+        'additionalProperties': False
+    }
+
+    def on_task_input(self, task, config):
+        url = '%s:%s/api/series' % (config['base_url'], config['port'])
+        headers = {'X-Api-Key': config['api_key']}
+        json = task.requests.get(url, headers=headers).json()
+        entries = []
+        for show in json:
```

Comment:
maybe a check could be added to see if the show is monitored in sonarr and hasn't ended by adding something like

    if show["monitored"] and show["status"] != "ended"

this would cut down on searching and would respect the sonarr settings